man/docker.1.md

```diff
@@ -425,6 +425,26 @@ device.
 
 Example use: `docker daemon --storage-opt dm.use_deferred_removal=true`
 
+#### dm.use_deferred_deletion
```

Comment:
@rhvgoyal looks useful. A few things, this section seems to repeat itself.  It would be more useful to restructure it a bit. Also, I think it is useful to include the error users encounter if this is *not* enabled.

Also, you need to add this same section to the `docs/reference/commandline/daemon.md` file

-->  https://gist.github.com/moxiegirl/b8bbf5b4c74ad2f35e9c

#### dm.use_deferred_deletion

Enables use of deferred device deletion for thin pool devices. By default, thin pool device deletion is synchronous. Before a container is deleted, the Docker daemon removes any associated devices. If the storage driver cannot remove a device, the container deletion fails and the daemon returns:

```
Error deleting container: Error response from daemon: Cannot destroy container
```

To avoid this failure, enable both deferred device deletion and deferred device
removal on the daemon:

`docker daemon --storage-opt dm.use_deferred_deletion=true --storage-opt
dm.use_deferred_removal=true`

With these two options enabled, if a device is busy when the driver is deleting
a container, the driver marks the device as deleted. Later, when the device
isn't in use, the driver deletes it.

You should enable deferred device deletion in situations where you expect to
mount the directory of one container into another.  In these situations, a mount
point can leak into multiple namespaces causing the device to be busy.